deflect/config.h

```diff
@@ -45,8 +45,10 @@
 #    define final
 #    define override
 #  endif
-#  ifndef CXX_NOEXCEPT_SUPPORTED
-#    define noexcept
+#  ifdef CXX_NOEXCEPT_SUPPORTED
+#    define NOEXCEPT noexcept
```

Comment:
Turns out it is somewhat illegal to (re)define C++ keywords:
http://stackoverflow.com/questions/9109377/is-it-legal-to-redefine-a-c-keyword

That's what gcc is not telling you, but MSVC gives a warning here. Would not be a problem if MSVC would support 'noexcept', but iirc even the latest version still has no support for it...